### PR TITLE
[doc] chore: npu docker support vllm013

### PR DESCRIPTION
**Pull request by yyyy2000**
### What does this PR do?

npu docker support vllm013

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: docker/ascend/Dockerfile.ascend_8.5.rc1_a3 [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Pull base image
+FROM swr.cn-south-1.myhuaweicloud.com/ascendhub/cann:8.5.0-a3-ubuntu22.04-py3.11
```
Comment (wucong25) [inline docker/ascend/Dockerfile.ascend_8.5.rc1_a3:2]: 切换vllm013修复的代码最好一起上仓

File: verl/models/mcore/patch.py [hunk 2]
```diff
@@ -362,3 +362,103 @@ def get_tensor_model_parallel_group_if_none(tp_group, is_expert=False, check_ini
             return tp_group
 
         megatron.core.utils.get_tensor_model_parallel_group_if_none = get_tensor_model_parallel_group_if_none
+
+
```
Comment (wucong25) [inline verl/models/mcore/patch.py:366]: 空行留得有点多 一般留一到两个空行

File: verl/models/mcore/patch.py [hunk 3]
```diff
@@ -362,3 +362,103 @@ def get_tensor_model_parallel_group_if_none(tp_group, is_expert=False, check_ini
...
+
+def apply_patch_012megatron_with_28torch():
```
Comment (wucong25) [inline verl/models/mcore/patch.py]: 命名方式apply_patch_megatron_v012_with_torch_v28感觉更加清晰一点

File: verl/models/mcore/patch.py [hunk 4]
```diff
@@ -362,3 +362,103 @@ def get_tensor_model_parallel_group_if_none(tp_group, is_expert=False, check_ini
...
+        version.parse(torch.__version__).base_version != "2.8.0"
+        or version.parse(megatron.core.__version__).base_version != "0.12.1"
+    ):
+        return
+
+    WriteBucket = tuple[Path, str, tuple[list, list]]
+
+    @staticmethod
+    @_disable_gc()
+    def write_preloaded_data_patch(
```
Comment (wucong25) [inline verl/models/mcore/patch.py:393]: 具体patch的地方建议添加注释

File: scripts/converter_hf_to_mcore.py [hunk 5]
```diff
@@ -592,7 +592,9 @@ def megatron_model_provider(pre_process, post_process):
 
     # save megatron model
     if len(os.listdir(output_path)) == 0:
-        dist_checkpointing.save(megatron_state_dict, output_path, sharded_strategy=None, async_sharded_save=False)
```
Comment (wucong25) [inline scripts/converter_hf_to_mcore.py:595]: 这一行为啥删除了，确认一下是否误删

File: verl/utils/megatron/dist_checkpointing.py [hunk 6]
```diff
@@ -50,7 +50,9 @@ def save_dist_checkpointing(
     if content_metadata is not None:
         if mcore_ge_014:
             save_kwargs["content_metadata"] = content_metadata
+    from verl.models.mcore.patch import apply_patch_megatron_v012_with_torch_v28
 
+    apply_patch_megatron_v012_with_torch_v28()
```
Comment (wucong25) [inline verl/utils/megatron/dist_checkpointing.py]: 这个patch是npu、gpu共性的么，如果只是npu上会遇到的话，建议放到npu_patch中

File: verl/models/mcore/patch.py (hunk 2 above)
Comment (yyyy2000) [reply verl/models/mcore/patch.py]: 已根据意见删除空行
